dll/shellext/zipfldr/CZipExtract.cpp

```diff
@@ -591,10 +592,20 @@ class CZipExtract :
 
             bool is_dir = Name.GetLength() > 0 && Name[Name.GetLength()-1] == '/';
 
-            char CombinedPath[MAX_PATH * 2] = { 0 };
-            PathCombineA(CombinedPath, BaseDirectory, Name);
-            CStringA FullPath = CombinedPath;
-            FullPath.Replace('/', '\\');    /* SHPathPrepareForWriteA does not handle '/' */
+            // Build a combined path
+            CStringW CombinedPath = BaseDirectory;
+            if (CombinedPath.GetLength() > 0 && CombinedPath[CombinedPath.GetLength() - 1] != L'\\')
+                CombinedPath += L'\\';
```

Comment:
This kind of stuff is what `CPathW` is for.

Reply:
Used `CPathW::Combine`.